### DISCUSSION TITLE
Hash check after download

### DISCUSSION THREAD

**Discussion by Git-Maniac** [BlafKing/sd-civitai-browser-plus#190] [Ideas]
Hi 
The civitai browser is for me one of the most used extensions for Auto1111! I use it almost every day! Thank you lots for that!!

I have noticed that after downloading from Civitai Browser+ large safetensors do not always work. Auto1111 reports various errors. If I delete it in Civitai Browser+ and download it again, it works fine. This is rare, but it does happen.

If others have experienced this, I suggest that after downloading from Civitai Browser+ the hash of the file is automatically checked or there is an option to do so. On the Civitai site all models have a hash. So it should be possible to integrate this.

At the moment I check the hash of every file with a separate program after downloading with Civitai Browser+.

What do you think?

**Comment by BlafKing** [comment]
That sounds like a good idea,
I'll add this to the to-do list!

When I get around to adding it I think I'll add it as a toggle in the SD-WebUI settings of this extension.
